// This program is free software, you can redistribute it and/or modify it.
// Copyright (c) 2025 Huawei Technologies Co., Ltd.
// This file is a part of the CANN Open Software.
// Licensed under CANN Open Software License Agreement Version 2.0 (the "License").
// Please refer to the License for details. You may not use this file except in compliance with the License.
// THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING
// BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
// See LICENSE in the root of the software repository for the full text of the License.

// The code snippet comes from Huawei's open-source Ascend project.
// Copyright 2020 Huawei Technologies Co., Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// You may obtain a copy of the License at 
// http://www.apache.org/licenses/LICENSE-2.0

syntax = "proto3";

option cc_enable_arenas = true;
import "cpu_tensor_shape.proto";
package aicpuops;

message Tensor {

  // tensor shape info
  TensorShape tensor_shape = 1;

  // tensor content data type
  int32 tensor_type = 2;

  // tensor memory device
  // data located memory device , "DDR" "HBM" OR "NONE"
  string mem_device = 3;
  string name = 4;
  uint64 data_ptr = 5;
  uint64 data_size = 6;
}
